CDP4JsonFileDal/JsonFileDal.cs

```diff
@@ -190,12 +218,15 @@ public override Task<IEnumerable<Thing>> Write(IEnumerable<OperationContainer> o
                 siteReferenceDataLibraries,
                 domainsOfExpertises,
                 persons,
+                personRoles,
+                participantRoles,
+                organizations,
                 engineeringModelSetups,
                 iterationSetups);
 
             var activePerson = JsonFileDalUtils.QueryActivePerson(this.Session.Credentials.UserName, siteDirectory);
 
-            var exchangeFileHeader = JsonFileDalUtils.CreateExchangeFileHeader(activePerson);
+            var exchangeFileHeader = this.FileHeader is ExchangeFileHeader ? this.FileHeader : JsonFileDalUtils.CreateExchangeFileHeader(activePerson);

```

Comment:
Might be simpler as `this.FileHeader as ExchangeFileHeader ?? JsonFileDalUtils.CreateExchangeFileHeader(...)`